feat: auto-close issues from merge commits to default branch

ludamad opened the pull request:
## Problem

When PRs target intermediate branches (e.g., merge-train) instead of the main branch, GitHub's native auto-close functionality doesn't work. Issues referenced with `Closes #123` remain open even after the PR eventually merges to `next`.


Recognizes cross-repo references and various `resolves` `fixes` etc

sirasistant reviewed (approved):
LGTM only nit from me is for it to pick the url form also not only the #issue number form

ludamad committed:
feat: auto-close issues from merge commits to default branch

## Problem

When PRs target intermediate branches (e.g., merge-train) instead of the main branch, GitHub's native auto-close functionality doesn't work. Issues referenced with `Closes #123` remain open even after the PR eventually merges to `next`.

Recognizes cross-repo references and various `resolves` `fixes` etc
